x/erc20/keeper/proposals.go

```diff
@@ -153,6 +155,39 @@ func (k Keeper) CreateCoinMetadata(
 	return &metadata, nil
 }
 
+// CreateERC20Metadata generates the metadata to represent the ERC20 token.
+func (k Keeper) CreateERC20Metadata(
+	ctx sdk.Context,
+	base, name, symbol string,
+	decimals uint32,
+) (*banktypes.Metadata, error) {
+	// create a bank denom metadata based on the ERC20 token ABI details
+	// metadata name is should always be the contract since it's the key
+	// to the bank store
+	metadata := banktypes.Metadata{
+		Description: types.CreateDenomDescription(""), // TODO: what description do we add here
+		Base:        base,
+		// NOTE: Denom units MUST be increasing
+		DenomUnits: []*banktypes.DenomUnit{
+			{
+				Denom:    base,
+				Exponent: 0,
+			},
+			{
+				Denom:    name,
+				Exponent: decimals,
+			},
+		},
+		Name:    name,
+		Symbol:  symbol,
+		Display: base,
+	}
+
+	k.bankKeeper.SetDenomMetaData(ctx, metadata)
```

Comment:
This is a simplified function for metadata without the need for a concrete SC. Does it look correct ?